pkg/kubelet/kubelet_pods.go

```diff
@@ -570,6 +570,18 @@ func (kl *Kubelet) GetOrCreateUserNamespaceMappings(pod *v1.Pod, runtimeHandler
 func (kl *Kubelet) GeneratePodHostNameAndDomain(pod *v1.Pod) (string, string, error) {
 	clusterDomain := kl.dnsConfigurer.ClusterDomain
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.HostnameOverride) && pod.Spec.HostnameOverride != nil {
+		hostname := *pod.Spec.HostnameOverride
+		if msgs := utilvalidation.IsDNS1123Label(hostname); len(msgs) != 0 {
+			return "", "", fmt.Errorf("pod HostnameOverride %q is not a valid DNS label: %s", hostname, strings.Join(msgs, ";"))
+		}
```

Comment:
I imagine this is a "can not happen" kind of thing, right? the field should be validated in the apiserver before, no?

Reply:
https://github.com/kubernetes/kubernetes/pull/132558#discussion_r2181135210

Reply:
I see, standalone kubelets